pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -130,6 +131,8 @@ func (a *index) workloadEntryWorkloadBuilder(
 	namespaces krt.Collection[*v1.Namespace],
 ) krt.TransformationSingle[*networkingclient.WorkloadEntry, model.WorkloadInfo] {
 	return func(ctx krt.HandlerContext, wle *networkingclient.WorkloadEntry) *model.WorkloadInfo {
+		// WLE can put labels in multiple places; normalize this
+		wle = serviceentry.ConvertClientWorkloadEntry(wle)
```

Comment:
isn't there a divergence, ambient combines both metadata labels and spec.labels. while serviceeentry controller or k8s service select based on only spec.labels

Reply:
this same logic is applied to those controllers too. See ConvertWorkloadEntry 


it also was applied to ambient before a refactoring